src/transaction/requests.rs

```diff
@@ -294,7 +294,176 @@ pub fn new_cleanup_request(key: impl Into<Key>, start_version: u64) -> kvrpcpb::
     req
 }
 
-dummy_impl_has_locks!(PrewriteResponse);
+impl AsRef<Key> for kvrpcpb::Mutation {
```

Comment:
How is this used? I generally don't like this sort of thing - I think it is better to be explicit about conversions rather than using AsRef/Deref, etc.

Reply:
It's for implementing AsRef<Key> for Mutation. Any better idea?

Reply:
Mutation is used for store_stream_for_keys, which requires it to implement `AsRef<Key>`

Reply:
Hmm, that is my fault I guess. Maybe we should use a different trait for that. I would leave it for now, but add a comment.

Reply:
Agree. A different trait seems better.